contracts/traderjoe/FarmLens.sol

```diff
@@ -0,0 +1,199 @@
+        uint256 priceFromWavaxDai = _getAvaxPrice(IJoePair(address(0xA389f9430876455C36478DeEa9769B7Ca4E3DDB1))); // 18
+
+        uint256 sumPrice = priceFromWavaxUsdt.add(priceFromWavaxUsdc).add(priceFromWavaxDai); // 18
+        uint256 avaxPrice = sumPrice / uint256(3); // 18
+        return avaxPrice; // 18
+    }
+
+    function _getAvaxPrice(IJoePair pair) public view returns (uint256) {
+        (uint256 reserve0, uint256 reserve1, ) = pair.getReserves();
+
+        if (pair.token0() == wavax) {
+            reserve1 = reserve1.mul(_tokenDecimalsMultiplier(pair.token1())); // 18
+            return (reserve1.mul(uint256(1e18))) / reserve0; // 18
+        } else {
+            reserve0 = reserve0.mul(_tokenDecimalsMultiplier(pair.token0())); // 18
+            return (reserve0.mul(uint256(1e18))) / reserve1; // 18
+        }
+    }
+
+    function getPriceInUSD(address tokenAddress) public view returns (uint256) {
+        return getAvaxPrice() / getPriceInAvax(tokenAddress); // 18
+    }
+
+    // Need to be aware of decimals here, not always 18, it depends on the token
+    function getPriceInAvax(address tokenAddress) public view returns (uint256) {
+        if (tokenAddress == wavax) {
+            return uint256(1e18);
+        }
+
+        IJoePair pair = IJoePair(joeFactory.getPair(tokenAddress, wavax));
+
+        (uint256 reserve0, uint256 reserve1, ) = pair.getReserves();
+        address token0Address = pair.token0();
+        address token1Address = pair.token1();
+
+        if (token0Address == wavax) {
+            reserve1 = reserve1.mul(_tokenDecimalsMultiplier(token1Address)); // 18
+            return (reserve1.mul(uint256(1e18))) / reserve0; // 18
+        } else {
+            reserve0 = reserve0.mul(_tokenDecimalsMultiplier(token0Address)); // 18
+            return (reserve0.mul(uint256(1e18))) / reserve1; // 18
+        }
+    }
+
+    function _tokenDecimalsMultiplier(address tokenAddress) public pure returns (uint256) {
+        uint256 decimalsNeeded = 18 - IJoeERC20(tokenAddress).decimals();
+        return uint256(1 * (10**decimalsNeeded));
+    }
+
+    function getReserveUSD(IJoePair pair) public view returns (uint256) {
+        address token0Address = pair.token0();
+        address token1Address = pair.token1();
+
+        (uint256 reserve0, uint256 reserve1, ) = pair.getReserves(); // reserve0, reserve1 are 18 decimals
+
+        uint256 token0PriceInAvax = getPriceInAvax(token0Address); // 18
+        uint256 token1PriceInAvax = getPriceInAvax(token1Address); // 18
+        uint256 token0ReserveUSD = (reserve0.mul(_tokenDecimalsMultiplier(token0Address))).mul(token0PriceInAvax).mul(
+            getAvaxPrice()
+        ); // 18.mul(18).mul(18) = 54 decimals
+        uint256 token1ReserveUSD = (reserve1.mul(_tokenDecimalsMultiplier(token1Address))).mul(token1PriceInAvax).mul(
+            getAvaxPrice()
+        ); // 54
```

Comment:
Might be wrong here.. but I think it should be reverve0 * avaxPrice / token0PriceInAvax